torch/_dynamo/variables/dicts.py

```diff
@@ -216,6 +216,8 @@ def unpack_var_sequence(self, tx):
     def get_key(cls, arg: VariableTracker):
         if isinstance(arg, TensorVariable) and arg.specialized_value is not None:
             return arg.specialized_value
+        elif isinstance(arg, SymNodeVariable):
+            return arg.evaluate_expr()
```

Comment:
Hmm, I think this will end up raising an exception if `arg` is an unbacked symint.

What do we want to have happen? Make the dict no longer const? Graph break?

Reply:
We should graph break in this case

Reply:
Graph break - I kind assumed guard_scalar will handle this, but it does not appear so. Let's add a test and try->catch->unimplemented this bad boy